@xen-orchestra/fs/src/abstract.js

```diff
@@ -95,26 +148,46 @@ export default class RemoteHandlerAbstract {
     }
     ;({ highWaterMark: this._highWaterMark, timeout: this._timeout = DEFAULT_TIMEOUT } = options)
 
-    const sharedLimit = limitConcurrency(options.maxParallelOperations ?? DEFAULT_MAX_PARALLEL_OPERATIONS)
-    this.closeFile = sharedLimit(this.closeFile)
-    this.copy = sharedLimit(this.copy)
-    this.getInfo = sharedLimit(this.getInfo)
-    this.getSizeOnDisk = sharedLimit(this.getSizeOnDisk)
-    this.list = sharedLimit(this.list)
-    this.mkdir = sharedLimit(this.mkdir)
-    this.openFile = sharedLimit(this.openFile)
-    this.outputFile = sharedLimit(this.outputFile)
-    this.read = sharedLimit(this.read)
-    this.readFile = sharedLimit(this.readFile)
-    this.rename = sharedLimit(this.rename)
-    this.rmdir = sharedLimit(this.rmdir)
-    this.truncate = sharedLimit(this.truncate)
-    this.unlink = sharedLimit(this.unlink)
-    this.write = sharedLimit(this.write)
-    this.writeFile = sharedLimit(this.writeFile)
-
     this._forget = coalesceCalls(this._forget)
     this._sync = coalesceCalls(this._sync)
+
+    this._applySafeGuards(options)
+  }
+
+  _applySafeGuards(options, { toLimit = WITH_LIMIT, toTimeout = WITH_TIMEOUT, toRetry = WITH_RETRY } = {}) {
```

Comment:
forget and sync should also be handled here
you can define a private method in js like this #applySafeGuards
And I think this method shouldn't have any parameters, since the dependencies of timeout/retry/limit can be tricky 
Bonus: it could be obtained by the remote options ( like the timeout )